aggregator_api/src/lib.rs

```diff
@@ -136,87 +140,113 @@ async fn post_task<C: Clock>(
+    // 32 byte task ID by taking SHA-256(VDAF verify key).
+    // https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-04#name-verification-key-requiremen
+    let task_id = TaskId::try_from(digest(&SHA256, &vdaf_verify_key_bytes).as_ref())
+        .map_err(|err| Error::new(err.to_string(), Status::BadRequest))?;
+
+    let vdaf_verify_keys = Vec::from([SecretBytes::new(vdaf_verify_key_bytes)]);
+
+    let aggregator_auth_tokens = match req.role {
+        Role::Leader => {
+            let encoded = req.aggregator_auth_token.as_ref().ok_or_else(|| {
+                Error::new(
+                    "aggregator acting in leader role must be provided an aggregator auth token"
+                        .to_string(),
+                    Status::BadRequest,
+                )
+            })?;
             let token_bytes = URL_SAFE_NO_PAD.decode(encoded).map_err(|err| {
                 Error::new(
-                    format!("Invalid base64 value for collector_auth_token: {err}"),
+                    format!("Invalid base64 value for aggregator_auth_token: {err}"),
                     Status::BadRequest,
                 )
             })?;
-            // TODO(#472): Each token in the PostTaskReq should indicate whether it is a bearer
-            // token or a DAP-Auth-Token. For now, assume the latter.
-            Vec::from([DapAuthToken::try_from(token_bytes)
-                .map(AuthenticationToken::DapAuth)
-                .map_err(|_| {
-                    Error::new(
-                        "Invalid HTTP header value in collector_auth_token".to_string(),
-                        Status::BadRequest,
-                    )
-                })?])
+            Vec::from([
+                // TODO(#472): Each token in the PostTaskReq should indicate whether it is a bearer
+                // token or a DAP-Auth-Token. For now, assume the latter.
+                DapAuthToken::try_from(token_bytes)
+                    .map(AuthenticationToken::DapAuth)
+                    .map_err(|_| {
+                        Error::new(
+                            "Invalid HTTP header value in aggregator_auth_token".to_string(),
+                            Status::BadRequest,
+                        )
+                    })?,
+            ])
+        }
+
+        Role::Helper => {
+            if req.aggregator_auth_token.is_some() {
+                return Err(Error::new(
+                    "aggregator acting in helper role cannot be given an aggregator auth token"
+                        .to_string(),
+                    Status::BadRequest,
+                ));
+            }
+            // TODO(#472): switch to generating bearer tokens by default
+            Vec::from([AuthenticationToken::DapAuth(random())])
+        }
+
+        role => {
+            return Err(Error::new(
+                format!("invalid role {role}"),
+                Status::BadRequest,
+            ))
         }
```

Comment:
Nit: In the spirit of DRY we could have a role check at the top of the function
```rust
if !matches(role, Role::Leader | Role::Helper) {
    /* same error logic */
}
```
Then subsequent matches over role can have the last case be `_ => unreachable!()`.

Reply:
Thanks, that is tidier